src/pyhf/contrib/viz/brazil.py

```diff
@@ -27,24 +27,124 @@ def plot_results(ax, mutests, tests, test_size=0.05):
+
+    line_color = kwargs.pop("color", "black")
+    ax.plot(mutests, cls_obs, color=line_color, label=r"$\mathrm{CL}_{s}$")
+
+    for idx, color in zip(range(5), 5 * [line_color]):
         ax.plot(
             mutests, cls_exp[idx], c=color, linestyle='dotted' if idx != 2 else 'dashed'
         )
     ax.fill_between(mutests, cls_exp[0], cls_exp[-1], facecolor='yellow')
     ax.fill_between(mutests, cls_exp[1], cls_exp[-2], facecolor='green')
-    ax.plot(mutests, [test_size] * len(mutests), c='red')
+
+    test_size_color = kwargs.pop("test_size_color", "red")
+    ax.plot(mutests, [test_size] * len(mutests), color=test_size_color)
     ax.set_ylim(0, 1)
 
-    ax.set_xlabel(r"$\mu$ (POI)")
-    ax.set_ylabel(r"$\mathrm{CL}_{s}$")
+    x_label = kwargs.pop("xlabel", r"$\mu$ (POI)")
+    y_label = kwargs.pop("ylabel", r"$\mathrm{CL}_{s}$")
+    ax.set_xlabel(x_label)
+    ax.set_ylabel(y_label)
+
+
+def plot_cls_components(
+    ax, mutests, tests, test_size=0.05, clb_only=False, clsb_only=False, **kwargs
+):
+    r"""
+    Plot the values of :math:`\mathrm{CL}_{s+b}` and :math:`\mathrm{CL}_{b}`
+    --- the components of the :math:`\mathrm{CL}_{s}` ratio --- on top of the
+    :math:`\mathrm{CL}_{s}` values for a series of hypothesis tests for various
+    POI values.
+
+    Example:
+
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> import pyhf
+        >>> import pyhf.contrib.viz.brazil
+        >>> pyhf.set_backend("numpy")
+        >>> model = pyhf.simplemodels.hepdata_like(
+        ...     signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0]
+        ... )
+        >>> observations = [51, 48]
+        >>> data = observations + model.config.auxdata
+        >>> poi_vals = np.linspace(0, 5, 41)
+        >>> results = [
+        ...     pyhf.infer.hypotest(
+        ...         test_poi, data, model, return_expected_set=True, return_tail_probs=True
+        ...     )
+        ...     for test_poi in poi_vals
+        ... ]
+        >>> fig, ax = plt.subplots()
+        >>> pyhf.contrib.viz.brazil.plot_cls_components(ax, poi_vals, results)
+
+    Args:
+        ax (:obj:`matplotlib.axes.Axes`): The matplotlib axis object to plot on.
+        mutests (:obj:`list` or :obj:`array`): The values of the POI where the
+          hypothesis tests were performed.
+        tests (:obj:`list` or :obj:`array`): The collection of :math:`p`-values
+          from the hypothesis tests.
+          ``tests`` is required to have the same structure as
+          :func:`pyhf.infer.hypotest`'s return when using ``return_expected_set=True``
+          and ``return_tail_probs=True``: a tuple of :math:`\mathrm{CL}_{s}`,
+          :math:`\left[\mathrm{CL}_{s+b}, \mathrm{CL}_{b}\right]`,
+          :math:`\mathrm{CL}_{s,\mathrm{exp}}` band.
+        test_size (:obj:`float`): The size, :math:`\alpha`, of the test.
+        clb_only (:obj:`bool`): Bool for plotting only the :math:`\mathrm{CL}_{b}`
```

Comment:
I believe this also plots CLs? "Only" sounds to me like it would only plot CLb, instead what the options does looks to be more like "skip CLsb".

Reply:
Yeah, at the moment all of these plot the `CLs` along with either both or one of the `CL_s+b` or `CL_b`. You're right though that there should be an option to plot only the components and not the `CLs`. Maybe I should change these from `only_X` to `no_X` args.